src/main.tsx

```
@@ -0,0 +1,17 @@
+import { StrictMode } from 'react';
+import { createRoot } from 'react-dom/client';
+import './index.css';
+import App from './App.tsx';
+import { ErrorBoundary } from './components/ErrorBoundary.tsx';
+
+const root = document.getElementById('root');
+
+if (root) {
```

here it is better to throw an error before calling render for type narrowing to avoid implicit program behavior